runtime/vm/vmthread.c

```diff
@@ -391,6 +391,12 @@ void threadCleanup(J9VMThread * vmThread, UDATA forkedByVM)
 		/* Safe to call this whether handleUncaughtException clears the exception or not */
 		internalExceptionDescribe(vmThread);
 	}
+
+	#if (JAVA_SPEC_VERSION >= 14)
+		/* Refresh java.lang.Thread interrupt value so it is acessible when vm thread is cleaned up. */
+		J9VMJAVALANGTHREAD_SET_DEADINTERRUPT(vmThread, vmThread->threadObject, omrthread_interrupted(vmThread->osThread));
```

Comment:
Can this be moved into the existing java "cleanUpAttachedThread" call?

The call path, starting below in `cleanUpAttachedThread` is:
https://github.com/eclipse/openj9/blob/ed6c69f07441718bb9ab8cbf65c4d755f9ca56c3/runtime/vm/callin.cpp#L459

https://github.com/eclipse/openj9/blob/906d4009533c1c313e9e31a9e5db895e9ff0dcdf/jcl/src/java.base/share/classes/java/lang/J9VMInternals.java#L318

https://github.com/eclipse/openj9/blob/5a2386c150d3277d326c05224b206ef041ffbc73/jcl/src/java.base/share/classes/java/lang/Thread.java#L1527

Could this `cleanup` method call `isInterrupted()` and handle this final set in java?  Using the java native to do the set would also clear the native thread's interrupted state which would be a plus, given we recycle the native threads at the vm level

Reply:
If we can't do it in java, then I'd like to see this moved as late as possible - so it's done in the same acquire / release block as the `cleanUpAttachedThread` call